src/js.cookie.mjs

```diff
@@ -82,9 +93,8 @@ function init (converter) {
       }
 
       try {
-        var name = decode(parts[0])
-        jar[name] =
-          (converter.read || converter)(cookie, name) || decode(cookie)
+        var name = rfc6265Converter.read(parts[0])
+        jar[name] = converter.read(cookie, name) || rfc6265Converter.read(cookie, name)
```

Comment:
More of note to myself: do we still need this particular fallback behavior? In the regular case `converter.read` will be  `rfc6265Converter.read`.. only if a custom read converter would return a falsy value we'd fallback to standard rfc6265 read conversion. If I remember correctly this had been implemented as a mechanism to write a converter that only acts on particular keys and otherwise forwards the value to the standard converter. But now that we're providing access to the standard converter, such a mechanism can easily implemented within the custom converter altogether...:

```javascript
var customReadConverter = (value, key) => {
  if (value === 'special") {
    return value..toUpperCase()
  }
  return Cookies.converter.read(value)
}
```

Reply:
I'm also all in for this, in fact this is the issue I had on my head when [I asked to hold release of 3.0](https://github.com/js-cookie/js-cookie/issues/565)

This is needed to pass thoses tests specifically:
https://github.com/js-cookie/js-cookie/blob/master/test/tests.js#L484
https://github.com/js-cookie/js-cookie/blob/master/test/tests.js#L490

This is particulary problematic from my point of view (typescript user) because it weaken the assumption you can make on return type.
If you pass a read converter of type `<T>` you would assume that the following
`
Cookie<T>{
  get(name: string): T | undefined;
  get(): {[key: string]: T};
}
`
But because of this fallback we have to assume that every get could return an union type `T | string`:
`
Cookie<T>{
  get(name: string): T | string | undefined;
  get(): {[key: string]: T | string};
}
`
It would be really great if this could be avoided.

Reply:
=>#571